doc/man5/flux-config-exec.rst

```diff
@@ -50,6 +50,16 @@ kill-signal
    (optional) Specify an alternate signal to ``SIGKILL`` when killing tasks
    and the job shell. Mainly used for testing.
 
+barrier-timeout
+   (optional) Specify the default start barrier timeout in FSD for all
+   jobs. All multi-node jobs enter a barrier at startup once initialization
+   on each node is complete and individual tasks have been launched. Once
+   the first node enters this barrier, the job execution system starts a
+   timer, and if the timer expires before the barrier is complete, then
+   a job exception is raised warning that some nodes did not reach the
+   barrier and may be hung. To disable the barrier timeout, set this value
+   to ``"0"``. (Default: ``30m``).
```

Comment:
It might be good to specifically call out the "flux shell barrier" to avoid confusion with the PMI barrier that comes up so often when troubleshooting MPI startup.

Reply:
Good idea. I have rewritten this section (there was also an error, the init barrier occurs before tasks are started, not after)